plugins/meta/isolation/main.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/coreos/go-iptables/iptables"
+
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+)
+
+// PluginName
+const PluginName = "isolation"
+
+// PluginConf
+type PluginConf struct {
+	types.NetConf
+
+	// Chained prev results
+	RawPrevResult *map[string]interface{} `json:"prevResult"`
+	PrevResult    *current.Result         `json:"-"`
+
+	// Internal states
+	bridgeName string
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse network configuration: %v", err)
+	}
+
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	} else {
+		return nil, fmt.Errorf("needs to be chained with \"bridge\" plugin")
+	}
+
+	if len(conf.PrevResult.Interfaces) == 0 {
+		return nil, fmt.Errorf("interface needs to be set, make sure to chain %q plugin with \"bridge\"", PluginName)
+	}
+	intf := conf.PrevResult.Interfaces[0]
```

Comment:
maybe this interface should be checked whether it is a bridge, but not a veth peer or others

Reply:
Not strictly made for bridges: https://github.com/containernetworking/plugins/pull/574#issuecomment-763312252